### PR TITLE
Add side-products of Python compilation to gitignore

### PR DESCRIPTION
**Pull request by lidizheng**
Let's try to make the directory (looks) clean after compilation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,6 +19,8 @@ py27_gevent/
 py27_native/
 py3[0-9]_gevent/
 py3[0-9]_native/
+a.out
+src/python/grpcio_*/LICENSE
```
Comment (gnossen) [inline .gitignore]: Does `src/python/grpcio` also create a license?

File: .gitignore [hunk 2]
```diff
@@ -19,6 +19,8 @@ py27_gevent/
 py27_native/
 py3[0-9]_gevent/
 py3[0-9]_native/
+a.out
```
Comment (gnossen) [inline .gitignore]: Where does this even come from?

Comment (lidizheng) [reply .gitignore]: My guess is the "compiler_ok_with_extra_std" function in our build process.

File: .gitignore (hunk 1 above)
Comment (lidizheng) [reply .gitignore]: Nope. It uses the root directory one.

File: .gitignore (hunk 2 above)
Comment (gnossen) [reply .gitignore]: Probably a good idea to also fix that. Adding a TODO for now would also be fine.

**Comment by lidizheng** [comment]
Known failure: https://github.com/grpc/grpc/issues/21301

**Pull request merged**